src/AlgoliaSearchCore.js

```diff
@@ -686,6 +710,13 @@ AlgoliaSearchCore.prototype._incrementCurrentHostIndex = function(hostType) {
   );
 };
 
+AlgoliaSearchCore.prototype._getTimeoutsForRequest = function(hostType, tries) {
+  return {
+    connect: this._timeouts.connect * tries,
```

Comment:
To my knowledge, timeouts gets double after each try. Here we have timeouts that are progressing linearly. Is it on purpose?

More details on what I understand if the connect timeout is 1:
 - here for each tries (3): 1, 2, 3
 - what I thought is "sorta" standard: 1, 2, 4

*update after talking with @clement-leprovost* It is the same in the swift counterpart :)